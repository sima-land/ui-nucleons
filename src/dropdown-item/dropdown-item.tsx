import type { DropdownItemProps } from './types';
import classNames from 'classnames/bind';
import styles from './dropdown-item.module.scss';

const cx = classNames.bind(styles);

/**
 * Элемент выпадающего списка.
 * @param props Свойства.
 * @todo Переименовать в MenuItem.
 * @return Элемент.
 */
export function DropdownItem({
  value,
  size = 'm',
  children,
  className,
  selected,
  disabled,
  noHover,
  checked,
  startContent,
  startIcon: StartIcon,
  endContent,
  endIcon: EndIcon,
  comment,
  dangerouslySetInnerHTML,
  'data-testid': testId = 'dropdown-item',
  ...restProps
}: DropdownItemProps) {
  const start = StartIcon ? <StartIcon className={cx('icon')} /> : startContent;
  const end = EndIcon ? <EndIcon className={cx('icon')} /> : endContent;

  const hasChildren = [start, children, end].some(item => typeof item !== 'undefined');

  return (
    <div
      {...(value !== undefined && { 'data-value': value })}
      {...restProps}
      className={cx(
        'root',
        `size-${size}`,
        selected && 'selected',
        checked && 'checked',
        disabled && 'disabled',
        noHover && 'no-hover',
        className,
      )}
      dangerouslySetInnerHTML={dangerouslySetInnerHTML}
      data-testid={testId}
    >
      {/*
        ВАЖНО: при hasChildren === false нужно выводить именно undefined чтобы не падала ошибка:
        Can only set one of `children` or `props.dangerouslySetInnerHTML`.
      */}
      {hasChildren ? (
        <>
          {start && <div className={cx('col', 'col-start')}>{start}</div>}

          <div className={cx('col', 'col-center')}>
            <div className={cx('row-main')}>{children}</div>
            {size === 'xl' && comment && <div className={cx('row-comment')}>{comment}</div>}
          </div>

          {end && <div className={cx('col', 'col-end')}>{end}</div>}
        </>
      ) : undefined}
    </div>
  );
}
