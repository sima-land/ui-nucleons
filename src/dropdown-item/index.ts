export type { DropdownItemProps, DropdownItemSize } from './types';
export { DropdownItem } from './dropdown-item';
