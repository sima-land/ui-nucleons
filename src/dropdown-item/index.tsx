import { ComponentType, ReactNode, SVGAttributes } from 'react';
import classnames from 'classnames/bind';
import styles from './dropdown-item.module.scss';

const cx = classnames.bind(styles);

export type DropdownItemSize = 's' | 'm' | 'l' | 'xl';

export interface DropdownItemProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'size'> {
  /** Отображать элемент как отмеченный (активный) или нет. */
  checked?: boolean;

  /** Отображать элемент как выбранный или нет. */
  selected?: boolean;

  /** Отключен ли элемент. */
  disabled?: boolean;

  /** Нужно ли отключить эффект при наведении. */
  noHover?: boolean;

  /** Размер. */
  size?: DropdownItemSize;

  /** Строковое значение, ассоциированное с элементом списка. */
  value?: string;

  /** Контент перед основным содержимым. */
  startContent?: ReactNode;

  /** Контент после основного содержимого. */
  endContent?: ReactNode;

  /** Иконка перед основным содержимым. При указании будет проигнорирован "startContent". */
  startIcon?: ComponentType<SVGAttributes<SVGSVGElement>>;

  /** Иконка после основного содержимого. При указании будет проигнорирован "endContent". */
  endIcon?: ComponentType<SVGAttributes<SVGSVGElement>>;

  /** Контент под основным содержимым. Выводится только при size='xl'. */
  comment?: ReactNode;

  /** Идентификатор для систем автоматизированного тестирования. */
  'data-testid'?: string;
}

/**
 * Элемент выпадающего списка.
 * @param props Свойства.
 * @return Элемент.
 */
export function DropdownItem({
  size = 'm',
  children,
  className,
  selected,
  disabled,
  noHover,
  checked,
  startContent,
  startIcon: StartIcon,
  endContent,
  endIcon: EndIcon,
  comment,
  dangerouslySetInnerHTML,
  'data-testid': testId = 'dropdown-item',
  ...restProps
}: DropdownItemProps) {
  const start = StartIcon ? <StartIcon className={cx('icon')} /> : startContent;
  const end = EndIcon ? <EndIcon className={cx('icon')} /> : endContent;

  const hasChildren = [start, children, end].some(item => typeof item !== 'undefined');

  return (
    <div
      {...restProps}
      className={cx(
        'root',
        `size-${size}`,
        selected && 'selected',
        checked && 'checked',
        disabled && 'disabled',
        noHover && 'no-hover',
        className,
      )}
      dangerouslySetInnerHTML={dangerouslySetInnerHTML}
      data-testid={testId}
    >
      {/*
        ВАЖНО: при hasChildren === true нужно выводить именно undefined чтобы не падала ошибка:
        Can only set one of `children` or `props.dangerouslySetInnerHTML`.
      */}
      {hasChildren ? (
        <>
          {start && <div className={cx('col', 'col-start')}>{start}</div>}

          <div className={cx('col', 'col-center')}>
            <div className={cx('row-main')}>{children}</div>
            {size === 'xl' && comment && <div className={cx('row-comment')}>{comment}</div>}
          </div>

          {end && <div className={cx('col', 'col-end')}>{end}</div>}
        </>
      ) : undefined}
    </div>
  );
}
