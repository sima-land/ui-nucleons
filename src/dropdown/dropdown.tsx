import type { DropdownProps } from './types';
import { CustomScrollbar } from '../_internal/custom-scrollbar';
import classnames from 'classnames/bind';
import styles from './dropdown.m.scss';

const cx = classnames.bind(styles);

/**
 * Выпадающий блок. Обычно используется для вывода списков опций выбора.
 * @param props Свойства.
 * @return Элемент.
 */
export function Dropdown({
  rootRef,
  viewportRef,
  viewportProps,
  className,
  children,
  'data-testid': testId = 'dropdown',
  ...restProps
}: DropdownProps) {
  return (
    <div ref={rootRef} className={cx('root', className)} {...restProps} data-testid={testId}>
      <CustomScrollbar
        className={cx('inner')}
        overflow={{ x: 'hidden', y: 'scroll' }}
        viewportRef={viewportRef}
        viewportProps={viewportProps}
      >
        {children}
      </CustomScrollbar>
    </div>
  );
}
