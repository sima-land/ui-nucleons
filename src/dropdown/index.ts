export type { DropdownProps, DropdownStyle } from './types';
export { Dropdown } from './dropdown';
