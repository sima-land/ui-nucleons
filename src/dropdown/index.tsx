import React, { forwardRef } from 'react';
import { BoxShadow } from '../styling/shadows';
import { MediumRounds } from '../styling/shapes';
import { CustomScrollbar, CustomScrollbarProps } from '../_internal/custom-scrollbar';
import classnames from 'classnames/bind';
import styles from './dropdown.module.scss';

interface DropdownStyle extends React.CSSProperties {
  '--dropdown-max-height'?: string;
}

export interface DropdownProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'style'> {
  /** Стили. */
  style?: DropdownStyle;

  /** Идентификатор для систем автоматизированного тестирования. */
  'data-testid'?: string;

  /** Свойства компонента CustomScrollbar. */
  customScrollbarProps?: Omit<CustomScrollbarProps, 'className' | 'overflow' | 'children'>;
}

const cx = classnames.bind(styles);

/**
 * Компонент выпадающего блока с контентом, например списком.
 */
export const Dropdown = forwardRef<HTMLDivElement | null, DropdownProps>(function Dropdown(
  { className, children, 'data-testid': testId = 'dropdown', customScrollbarProps, ...restProps },
  ref,
) {
  return (
    <div
      ref={ref}
      className={cx('root', className, BoxShadow.z4, MediumRounds.all)}
      {...restProps}
      data-testid={testId}
    >
      <CustomScrollbar
        {...customScrollbarProps}
        className={cx('inner')}
        overflow={{ x: 'h', y: 's' }}
      >
        {children}
      </CustomScrollbar>
    </div>
  );
});
