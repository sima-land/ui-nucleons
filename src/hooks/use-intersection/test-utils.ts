import { act } from '@testing-library/react';

/**
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export type IntersectionObserverEntryBase = Omit<
  IntersectionObserverEntry,
  'time' | 'rootBounds' | 'boundingClientRect' | 'intersectionRect'
>;

let intersectionMock: IntersectionMock | undefined;

/**
 * Возвращает новый объект, частично реализующий IntersectionObserverEntry.
 * @param base Базовая информация.
 * @return Объект.
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export function createFakeEntry(base: IntersectionObserverEntryBase): IntersectionObserverEntry {
  const entry: IntersectionObserverEntry = {
    ...base,

    time: Date.now(),

    /** @inheritdoc */
    get rootBounds() {
      // eslint-disable-next-line no-console
      console.error('Not implemented in mock');
      return null as any;
    },

    /** @inheritdoc */
    get boundingClientRect() {
      // eslint-disable-next-line no-console
      console.error('Not implemented in mock');
      return null as any;
    },

    /** @inheritdoc */
    get intersectionRect() {
      // eslint-disable-next-line no-console
      console.error('Not implemented in mock');
      return null as any;
    },
  } as unknown as IntersectionObserverEntry;

  return entry;
}

/**
 * Реализует интерфейс для управления состоянием пересечения элемента с область видимости.
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export class IntersectionMock {
  _original: typeof IntersectionObserver;
  _observers: FakeIntersectionObserver[] = [];

  /** Конструктор. */
  constructor() {
    this._original = window.IntersectionObserver;
  }

  /**
   * Подменяет глобальный конструктор IntersectionObserver.
   */
  apply() {
    if (intersectionMock) {
      throw Error('Already mocked');
    }

    // eslint-disable-next-line @typescript-eslint/no-this-alias
    intersectionMock = this;
    window.IntersectionObserver = FakeIntersectionObserver;
  }

  /**
   * Восстанавливает оригинальный глобальный конструктор IntersectionObserver.
   */
  restore() {
    if (window.IntersectionObserver !== FakeIntersectionObserver) {
      throw Error('Not mocked');
    }

    window.IntersectionObserver = this._original;
    intersectionMock = undefined;
  }

  /**
   * Имитирует изменение состояния пересечения элемента с областью видимости.
   * @param entryArg Данные пересечения элемента.
   */
  changeElementState(entryArg: IntersectionObserverEntryBase) {
    act(() => {
      const entry = createFakeEntry(entryArg);

      for (const observer of this._observers) {
        if (
          observer._elements.some(e => e === entry.target) &&
          observer.thresholds.includes(entry.intersectionRatio)
        ) {
          (observer as FakeIntersectionObserver)._runCallback(entry);
        }
      }
    });
  }

  /**
   * Записывает observer в список зарегистрированных.
   * @param observer Observer.
   */
  _registerObserver(observer: FakeIntersectionObserver) {
    this._observers.push(observer);
  }

  /**
   * Удаляет observer из списка зарегистрированных.
   * @param observer Observer.
   */
  _unregisterObserver(observer: FakeIntersectionObserver) {
    this._observers.splice(this._observers.indexOf(observer), 1);
  }
}

/**
 * Имитация IntersectionObserver.
 * @deprecated Следует использовать утилиты из @sima-land/ui-nucleons/test-utils.
 */
export class FakeIntersectionObserver implements IntersectionObserver {
  readonly root: Element | Document | null;
  readonly rootMargin: string;
  readonly thresholds: ReadonlyArray<number>;

  _callback: IntersectionObserverCallback;
  _elements: Element[];

  /** @inheritdoc */
  constructor(callback: IntersectionObserverCallback, options: IntersectionObserverInit = {}) {
    this.root = options?.root || null;
    this.rootMargin = options?.rootMargin || '0px 0px 0px 0px';
    this.thresholds = Array.isArray(options.threshold)
      ? options.threshold
      : [options.threshold || 0];

    this._callback = callback;
    this._elements = [];
    intersectionMock?._registerObserver(this);
  }

  /** @inheritdoc */
  observe(element: Element) {
    this._elements.push(element);
  }

  /** @inheritdoc */
  unobserve(element: Element) {
    this._elements = this._elements.filter(e => e !== element);
  }

  /** @inheritdoc */
  disconnect() {
    this._elements = [];
    intersectionMock?._unregisterObserver(this);
  }

  /** @inheritdoc */
  takeRecords() {
    // eslint-disable-next-line no-console
    console.error('Not implemented in mock');

    return [];
  }

  /**
   * Запускает callback.
   * @param entry Данные пересечения элемента.
   */
  _runCallback(entry: IntersectionObserverEntry) {
    this._callback([entry], this);
  }
}
