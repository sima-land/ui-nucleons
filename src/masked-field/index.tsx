import {
  forwardRef,
  useEffect,
  useImperativeHandle,
  useRef,
  useState,
  FocusEvent as ReactFocusEvent,
} from 'react';
import { TextField, TextFieldProps } from '../text-field';
import { InputMask } from '@krutoo/input-mask/dist/dom';

export interface MaskState {
  value: string;
  cleanValue: string;
}

export interface MaskedFieldProps extends Omit<TextFieldProps, 'value' | 'onBlur' | 'multiple'> {
  /** Маска. */
  mask: string;

  /** Значение. */
  value?: string;

  /** Обработчик blur. */
  onBlur?: (e: ReactFocusEvent<HTMLInputElement>, s: MaskState) => void;
}

const maskCommons = { placeholder: '_', pattern: /\d/ };

/**
 * Текстовое поле (TextField) с маской.
 * @deprecated Теперь нужно использовать MaskedInput.
 * @param props Свойства.
 * @param props.mask Маска.
 * @param props.value Значение.
 * @param props.onBlur Сработает при событии blur.
 */
export const MaskedField = forwardRef<HTMLInputElement | undefined, MaskedFieldProps>(
  function MaskedField({ mask, value = '', onBlur, ...restOptions }, ref) {
    const [inputMask, setInputMask] = useState<any>();
    const [inputState, setInputState] = useState<MaskState>({ value, cleanValue: '' });
    const innerRef = useRef<HTMLInputElement>();

    useImperativeHandle(ref, () => innerRef.current);

    useEffect(() => {
      const im = InputMask(innerRef.current as any, {
        ...maskCommons,
        mask,
        onChange: setInputState,
      });

      im.setValue(value as string);
      setInputMask(im);

      return () => im.disable();
    }, [mask]);

    useEffect(() => {
      inputMask && inputMask.setValue(value);
    }, [value, inputMask]);

    return (
      <TextField
        {...restOptions}
        ref={innerRef}
        multiline={false}
        restPlaceholder={{
          value: mask.slice(inputState.value.length),
          shiftValue: inputState.value,
        }}
        // получаем данные маски только по событию "blur"
        onBlur={e => onBlur?.(e, inputState)}
      />
    );
  },
);
