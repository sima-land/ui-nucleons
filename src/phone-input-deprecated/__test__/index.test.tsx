import React from 'react';
import { mount } from 'enzyme';
import { act, Simulate } from 'react-dom/test-utils';
import { PhoneInput } from '..';
import { TextField } from '../../text-field';
import { Dropdown } from '../../dropdown';
import { DropdownItem } from '../../dropdown-item';
import { MaskedField } from '../../masked-field';
import { countries } from '../../phone-input/presets';
import { fireEvent, render } from '@testing-library/react';

describe('<PhoneInput />', () => {
  it('should render without props', () => {
    const wrapper = mount(<PhoneInput />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct countries in dropdown', () => {
    const wrapper = mount(<PhoneInput />);
    expect(wrapper.find(Dropdown)).toHaveLength(0);

    act(() => {
      wrapper.find('[data-testid="phone-input:dropdown-opener"]').simulate('click');
    });
    wrapper.update();
    expect(wrapper.find(Dropdown)).toHaveLength(1);
  });

  it('should call onCountrySelect on change country', () => {
    const onCountrySelect = jest.fn();
    const wrapper = mount(<PhoneInput onCountrySelect={onCountrySelect} />);

    expect(onCountrySelect).not.toHaveBeenCalled();
    wrapper.find('[data-testid="phone-input:dropdown-opener"]').simulate('click');
    expect(wrapper.find(Dropdown)).toHaveLength(1);

    act(() => {
      (wrapper.find(DropdownItem).at(0).prop('onClick') as any)();
    });
    wrapper.update();

    expect(wrapper.find(Dropdown)).toHaveLength(0);
    expect(onCountrySelect).toHaveBeenCalledTimes(1);
  });

  it('should open and close countries popup properly', () => {
    const wrapper = mount(<PhoneInput />);

    expect(wrapper.find(Dropdown)).toHaveLength(0);

    act(() => {
      wrapper.find('[data-testid="phone-input:dropdown-opener"]').simulate('click');
    });
    wrapper.update();
    expect(wrapper.find(Dropdown)).toHaveLength(1);

    // outside click
    act(() => {
      document.documentElement.dispatchEvent(new MouseEvent('click'));
    });
    wrapper.update();
    expect(wrapper.find(Dropdown)).toHaveLength(0);
  });

  it('should pass mask state to onBlur callback', () => {
    const spy = jest.fn();

    const wrapper = mount(<PhoneInput onBlur={spy} />);

    expect(spy).toHaveBeenCalledTimes(0);

    act(() => {
      wrapper.find(TextField).find('input').simulate('blur');
    });

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy.mock.calls[0][0].type).toBe('blur');
    expect(spy.mock.calls[0][1].cleanValue).toBe('7');
  });

  it('should define initial mask properly', () => {
    const wrapper = mount(<PhoneInput value='375112223344' />);

    expect(wrapper.find(MaskedField).prop('mask')).toBe(countries.belarus.mask);
  });

  it('should handle "value" prop change', () => {
    const spy = jest.fn();

    const wrapper = mount(<PhoneInput value='375112223344' onBlur={spy} />);

    act(() => {
      wrapper.find(TextField).find('input').simulate('blur');
    });

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy.mock.calls[0][0].type).toBe('blur');
    expect(spy.mock.calls[0][1].cleanValue).toBe('375112223344');

    wrapper.setProps({ value: '375000000000' });

    act(() => {
      wrapper.find(TextField).find('input').simulate('blur');
    });

    expect(spy).toHaveBeenCalledTimes(2);
    expect(spy.mock.calls[1][0].type).toBe('blur');
    expect(spy.mock.calls[1][1].cleanValue).toBe('375000000000');
  });

  it('Other: should render just text field with able to input only digits', () => {
    const changeSpy = jest.fn();
    const blurSpy = jest.fn();

    const wrapper = mount(<PhoneInput value='0000000' onBlur={blurSpy} onChange={changeSpy} />);

    expect(wrapper).toMatchSnapshot();
    expect(blurSpy).toHaveBeenCalledTimes(0);
    expect(changeSpy).toHaveBeenCalledTimes(0);

    act(() => {
      Simulate.change(wrapper.find('input').getDOMNode(), {
        target: { value: '000sss111' } as HTMLInputElement,
      });
    });

    expect(blurSpy).toHaveBeenCalledTimes(0);
    expect(changeSpy).toHaveBeenCalledTimes(1);
    expect(changeSpy.mock.calls[0][0].target.value).toBe('000111');

    act(() => {
      Simulate.blur(wrapper.find('input').getDOMNode(), {
        target: { value: '000111' } as HTMLInputElement,
      });
    });

    expect(blurSpy).toHaveBeenCalledTimes(1);
    expect(changeSpy).toHaveBeenCalledTimes(1);
    expect(blurSpy.mock.calls[0][1].ready).toBe(true);
    expect(blurSpy.mock.calls[0][1].value).toBe('000111');
    expect(blurSpy.mock.calls[0][1].cleanValue).toBe('000111');
  });

  it('should handle "label"', () => {
    const { getByTestId } = render(<PhoneInput value='0000000' label='My Phone' />);

    expect(getByTestId('text-field:label').textContent).toBe('My Phone');
  });

  it('should handle field block click: opener click case', () => {
    const { getByTestId } = render(<PhoneInput value='' onBlur={jest.fn()} />);

    const event = new MouseEvent('click', { bubbles: true });
    jest.spyOn(event, 'preventDefault');

    expect(event.preventDefault).toHaveBeenCalledTimes(0);
    fireEvent(getByTestId('phone-input:dropdown-opener'), event);
    expect(event.preventDefault).toHaveBeenCalledTimes(1);
  });

  it('should handle field block click: inside opener click case', () => {
    const { container } = render(<PhoneInput value='' onBlur={jest.fn()} />);

    const event = new MouseEvent('click', { bubbles: true });
    jest.spyOn(event, 'preventDefault');

    expect(event.preventDefault).toHaveBeenCalledTimes(0);
    fireEvent(
      container.querySelector('[data-testid="phone-input:dropdown-opener"] > img') as any,
      event,
    );
    expect(event.preventDefault).toHaveBeenCalledTimes(1);
  });

  it('should handle field block click: block click case', () => {
    const { getByTestId, queryAllByTestId } = render(<PhoneInput value='' onBlur={jest.fn()} />);

    // show dropdown menu
    expect(queryAllByTestId('phone-input:dropdown')).toHaveLength(0);
    fireEvent.click(getByTestId('phone-input:dropdown-opener'));
    expect(queryAllByTestId('phone-input:dropdown')).toHaveLength(1);

    const event = new MouseEvent('click', { bubbles: true });
    jest.spyOn(event, 'preventDefault');

    expect(queryAllByTestId('phone-input:dropdown')).toHaveLength(1);
    expect(event.preventDefault).toHaveBeenCalledTimes(0);
    fireEvent(getByTestId('text-field:block'), event);
    expect(queryAllByTestId('phone-input:dropdown')).toHaveLength(0);
    expect(event.preventDefault).toHaveBeenCalledTimes(0);
  });

  it('should open and close dropdown by click on opener', () => {
    const { getByTestId, queryAllByTestId } = render(<PhoneInput value='' onBlur={jest.fn()} />);

    // show dropdown menu
    expect(queryAllByTestId('phone-input:dropdown')).toHaveLength(0);
    fireEvent.click(getByTestId('phone-input:dropdown-opener'));
    expect(queryAllByTestId('phone-input:dropdown')).toHaveLength(1);

    fireEvent.click(getByTestId('phone-input:dropdown-opener'));
    expect(queryAllByTestId('phone-input:dropdown')).toHaveLength(0);
  });

  it('should display value by default when country is unknown', () => {
    const initialValue = '55566667777';

    const { getByTestId } = render(<PhoneInput value={initialValue} onBlur={jest.fn()} />);

    expect((getByTestId('text-field:field') as HTMLInputElement).value).toBe(initialValue);
  });
});
