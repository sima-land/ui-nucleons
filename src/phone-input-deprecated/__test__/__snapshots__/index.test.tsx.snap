// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneInput /> Other: should render just text field with able to input only digits 1`] = `
<PhoneInput
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  value="0000000"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      className="field"
      endAdornment={
        <div
          className="adornment"
          data-testid="phone-input:dropdown-opener"
          onClick={[Function]}
        >
          <img
            alt=""
            height={24}
            src="other.png"
            width={24}
          />
          <ForwardRef(DownSVG)
            className="arrow"
          />
        </div>
      }
      label="Телефон"
      multiline={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      value="0000000"
    >
      <div
        className="root field"
      >
        <div
          className="reset input-block variant-desktop size-l with-label rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
        >
          <div
            className="main-column"
          >
            <label
              className="label variant-desktop filled"
              data-testid="text-field:label"
            >
              Телефон
            </label>
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                value="0000000"
              >
                <div
                  className="reset root base-input variant-desktop size-l"
                  data-testid="base-input:root"
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    value="0000000"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
          <Box
            alignItems="center"
            display="flex"
            marginLeft={4}
          >
            <div
              className="display-flex align-items-center M-l__4"
            >
              <div
                className="adornment"
                data-testid="phone-input:dropdown-opener"
                onClick={[Function]}
              >
                <img
                  alt=""
                  height={24}
                  src="other.png"
                  width={24}
                />
                <ForwardRef(DownSVG)
                  className="arrow"
                >
                  <svg
                    className="arrow"
                    height={16}
                    viewBox="0 0 16 16"
                    width={16}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M3.247 6.342a1 1 0 0 1 1.412-.095L8 9.171l3.341-2.924a1 1 0 0 1 1.318 1.506l-4 3.5a1 1 0 0 1-1.317 0l-4-3.5a1 1 0 0 1-.095-1.411Z"
                      fillRule="evenodd"
                    />
                  </svg>
                </ForwardRef(DownSVG)>
              </div>
            </div>
          </Box>
        </div>
      </div>
    </ForwardRef(TextField)>
  </div>
</PhoneInput>
`;

exports[`<PhoneInput /> should render without props 1`] = `
<PhoneInput>
  <div
    className="root"
  >
    <ForwardRef(MaskedField)
      className="field"
      endAdornment={
        <div
          className="adornment"
          data-testid="phone-input:dropdown-opener"
          onClick={[Function]}
        >
          <img
            alt=""
            height={24}
            src="russia.png"
            width={24}
          />
          <ForwardRef(DownSVG)
            className="arrow"
          />
        </div>
      }
      label="Телефон"
      mask="+7 (___) ___-__-__"
      multiline={false}
      onBlur={[Function]}
      onClick={[Function]}
      value=""
    >
      <ForwardRef(TextField)
        className="field"
        endAdornment={
          <div
            className="adornment"
            data-testid="phone-input:dropdown-opener"
            onClick={[Function]}
          >
            <img
              alt=""
              height={24}
              src="russia.png"
              width={24}
            />
            <ForwardRef(DownSVG)
              className="arrow"
            />
          </div>
        }
        label="Телефон"
        multiline={false}
        onBlur={[Function]}
        onClick={[Function]}
        restPlaceholder={
          {
            "shiftValue": "+7 (",
            "value": "___) ___-__-__",
          }
        }
      >
        <div
          className="root field"
        >
          <div
            className="reset input-block variant-desktop size-l with-label rounds-sm__all"
            data-testid="text-field:block"
            onClick={[Function]}
          >
            <div
              className="main-column"
            >
              <label
                className="label variant-desktop filled"
                data-testid="text-field:label"
              >
                Телефон
              </label>
              <div
                className="input-wrapper"
              >
                <ForwardRef(BaseInput)
                  className="base-input variant-desktop size-l"
                  data-testid="text-field:field"
                  multiline={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  restPlaceholder={
                    {
                      "shiftValue": "+7 (",
                      "value": "___) ___-__-__",
                    }
                  }
                >
                  <div
                    className="reset root base-input variant-desktop size-l"
                    data-testid="base-input:root"
                  >
                    <span
                      aria-hidden={true}
                      className="fake-text"
                    >
                      <span
                        className="invisible-value"
                      >
                        +7 (
                      </span>
                      <span
                        className="placeholder"
                      >
                        ___) ___-__-__
                      </span>
                    </span>
                    <input
                      className="field reset"
                      data-testid="text-field:field"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onInput={[Function]}
                    />
                  </div>
                </ForwardRef(BaseInput)>
              </div>
            </div>
            <Box
              alignItems="center"
              display="flex"
              marginLeft={4}
            >
              <div
                className="display-flex align-items-center M-l__4"
              >
                <div
                  className="adornment"
                  data-testid="phone-input:dropdown-opener"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    height={24}
                    src="russia.png"
                    width={24}
                  />
                  <ForwardRef(DownSVG)
                    className="arrow"
                  >
                    <svg
                      className="arrow"
                      height={16}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clipRule="evenodd"
                        d="M3.247 6.342a1 1 0 0 1 1.412-.095L8 9.171l3.341-2.924a1 1 0 0 1 1.318 1.506l-4 3.5a1 1 0 0 1-1.317 0l-4-3.5a1 1 0 0 1-.095-1.411Z"
                        fillRule="evenodd"
                      />
                    </svg>
                  </ForwardRef(DownSVG)>
                </div>
              </div>
            </Box>
          </div>
        </div>
      </ForwardRef(TextField)>
    </ForwardRef(MaskedField)>
  </div>
</PhoneInput>
`;
